src/EventListener.php

```diff
@@ -107,6 +107,16 @@ public function completeSendingRequest(Request $request, Stream $stream): Promis
      */
     public function startReceivingResponse(Request $request, Stream $stream): Promise;
 
+    /**
+     * Called by {@see Stream::request()} when a 1xx informational response is received.
+     *
+     * @param Request  $request
+     * @param Response $response
+     *
+     * @return Promise Should resolve successfully, otherwise aborts the request.
+     */
+    public function handleInformationalResponse(Request $request, Response $response): Promise;
```

Comment:
This will be a BC break. My suggestion is to add it as `Request::setInformationalResponseHandler`, like `setPushHandler` and `setUpgradeHandler`. Maybe we can find a slightly shorter name for it.